src/Nethermind/Nethermind.Synchronization/StateSync/StateSyncDispatcher.cs

```diff
@@ -32,11 +38,42 @@ public StateSyncDispatcher(ISyncFeed<StateSyncBatch> syncFeed, ISyncPeerPool syn
         {
         }
 
-        protected override async Task Dispatch(PeerInfo peerInfo, StateSyncBatch request, CancellationToken cancellationToken)
+        protected override async Task Dispatch(PeerInfo peerInfo, StateSyncBatch batch, CancellationToken cancellationToken)
         {
+            if (batch?.RequestedNodes == null || batch.RequestedNodes.Length == 0)
+            {
+                return;
```

Comment:
Might wanna throw in this case, or make the feed return null. Otherwise, `SyncDispatcher` might just loop to this code path again immediately, causing a temporary CPU usage spike.